Frameworks/CoreGraphics/CGContext.mm

```diff
@@ -471,21 +475,32 @@ inline void ClearPath() {
     }
 
     inline bool ShouldDraw() {
-        return CurrentGState().ShouldDraw();
+        return FAILED(_firstErrorHr) || CurrentGState().ShouldDraw();
     }
 
     inline void PushBeginDraw() {
-        if ((_beginEndDrawDepth)++ == 0) {
+        if (!FAILED(_firstErrorHr) && (_beginEndDrawDepth)++ == 0) {
             deviceContext->BeginDraw();
         }
     }
 
     inline HRESULT PopEndDraw() {
+        HRESULT hr = S_OK;
         if (--(_beginEndDrawDepth) == 0) {
-            RETURN_IF_FAILED(deviceContext->EndDraw());
+            hr = deviceContext->EndDraw();
```

Comment:
i'm thinking about S_FALSE, looking at the documentation, i don't see much about it for EndDraw(), but given it's technically a success, I assume we don't have to worry about it?